apps/web/src/app/(profile)/[username]/_components/dashboard/badges/_difficulty_badges.ts

```diff
@@ -0,0 +1,58 @@
+import { prisma } from '@repo/db';
+import type {AllBadgeObjs, BadgeFn} from "~/app/(profile)/[username]/_components/dashboard/_actions";
+import type {SolutionBadges} from "~/app/(profile)/[username]/_components/dashboard/badges/_shared_solutions_badges";
+import type {AotBadges} from "~/app/(profile)/[username]/_components/dashboard/badges/_advent_badges";
+export const DifficultyBadgeKeys = ['BEGINNER', 'EASY', 'MEDIUM', 'HARD', 'EXTREME'] as const;
+
+export type DifficultyBadges = typeof DifficultyBadgeKeys[number];
+export interface Difficulty {
+  Difficulty: DifficultyBadges;
+  TotalCompleted: number;
+}
+
+export const difficultyBadgesFn: BadgeFn =
+  async ({
+           userId,
+           badges,
+         }: {
+    userId: string;
+    badges: AllBadgeObjs;
+  }) => {
+    // DifficultyBadges
+    const difficulty: Difficulty[] | null = await DifficultyRetrieveData(userId);
+    return await DifficultyBadgesFn(badges, difficulty ?? []);
```

Comment:
`DifficultyBadgesFn` vs `difficultyBadgesFn` is terribly confusing

Reply:
extremely - I've changed the naming on functions to fix this.